### ISSUE TITLE
Deploy / Test AIMS plugin for DAAS upgrade of QGIS (2.18) 

### ISSUE THREAD

**Issue by SPlanzer** [linz/QGIS-AIMS-Plugin#216]
@splanzer to oversee plugin release and testing for DASS QGIS upgrade with the support of the Addressing Team

**Comment by SPlanzer** [comment]
**What needs to be done?**
A new release of the AIMS QGIS Plugn must be made avialble for conducting testing agaisnt QGIS 2.18 
 as this is about to be deployed in DAAS (as per issues [171](https://github.com/linz/LI-tasks/issues/171))

As well as testing, this release will involve some development. This is as after the code freeze pre AIMS go live, minor bug fixes and enhancements were made in anticipation for the next release. This is to be the first release since go live in Nov 2016.

This coming release will address the below:
* #198 Map: Pending Parcels
* #195 Refresh: Accepting last item on queue
* #194 Map - Road Parcels
* #200 Display appellation layer

All but "#200 Display appellation layer" are ready for testing. This displaying of the appellation from the QGIS Plugin code prespective is ready to go but this requires some database development to supprt QGIS in displaying the labels. 

The requirement for this is to display the parcels associated appellation string. This appellation is got via several database functions that require bde.crs_appellation data. This data is noy currently deployed as part of the [AIMS deployment](https://github.com/linz/AIMS-deployment) and needs to be added

I have sought feed back and these enhancements and these are still demend of the highest prioirity for the addressign team. 

**What does DONE look like**
* Deploy QGIS Plugin
- [x] Create Release on GitHub 
- [x] Upload to LINZ Plugin repo

* Update AIMS Deplyoment
 - [x] The parcel appelation view must be installed as part of the AIMS deployment 
  * subset data 
  - [x] Update python-subset AIMS yaml config to include crs_appellation data
  - [x] Dump crs_appellation data using updated python-subset
  - [x] Update the AIMS s3 Store with newly dumped subset which includes crs_appellation data
  * Full data dump 
  - [x] The full data dump and restore must be updated to included the crs_appellation data

* Testing
- [x] Splanzer to conduct sanity test
- [ ] The addressing team to do a full regression test in QGIS 2.18
- [ ] The addressing team to sign of on the new functionality and bug fixes

* Deployment to production at the complettion of testing
- [ ] The view that QGIS depends on must be deployed to production.

**Comment by SPlanzer** [comment]
FYI @linzbjones

**Comment by linzbjones** [comment]
@jason-bedford the AIMS plugin and Roads plugin are nearly ready for release, can you please review AIMS test plan? Once reviewed get @SPlanzer @dwsilk to review it also.

**Comment by jason-bedford** [comment]
@linzbjones Where do I find the test plan? Is it in github?

**Comment by linzbjones** [comment]
@jason-bedford sent via email

**Comment by jason-bedford** [comment]
@linzbjones I've finished reviewing the AIMS test plan. 
I've saved it as a new spreadsheet named 'AIMS Test Plan' in the same folder as the old one (which is named AIMS 1 UI Test), as the old had contained 12 tabs full of stuff and was very confusing.
The new test plan is based on the tab named 'Test_Results' in the old plan.
I've hidden a bunch of columns in the new one to make it easier to read, but didn't delete them in case they may be needed in future.
One of the hidden tabs is named 'Old Test Number', which refers to the test number in the old test plan, in case we need to refer back.
I renumbered the tests in the new plan to give them a better flow.
Can @SPlanzer and @dwsilk also review it, and add anything that I may have missed.
I can email the link if you like.

**Comment by SPlanzer** [comment]
Thanks @jason-bedford. I am looking at this task right now. Once we have gone over the test plan it would be good to get some of your time to start testing the AIMS plugin. 

And, yes please, can you send a link to the test plan.

**Comment by linzbjones** [comment]
thanks, @jason-bedford. I will send the latest test plan to @SPlanzer and @dwsilk so they can look over it for you.

**Comment by SPlanzer** [comment]
AIMS stood up on LI-Server ready for testing.

Note (To myself and reviewers): This has been stood up via the app_view-plugin_labelling branch of AIMS-deployment and is awaiting a PR review [#57](https://github.com/linz/AIMS-deployment/pull/57) 

This update to the AIMS-Deployment sees a view added to the BDE schema to support table labeling.

**Comment by SPlanzer** [comment]
@jason-bedford I will review the test plan this morning. 

In the mean time, instructions on connecting to the test instance of the plugin and upgrading to the Release Candidate version of the AIMS plugin follow. I would be happy if you would like to grab me when you perform these steps initially and I can go through them with you to ensure it is painless as possible and add any extra detail not in here.

**A note about environments** 
The primary objective of this is to sign off on the functioning of the AIMS plugin against QGIS2.18 operating in Daas AND sign off on new functionality (List above and I will ensure it is listed in the test plan)

However, Right now you can not connect from DaaS straight to the LINZ-Experimental server, our test environment. You must be using an internet connection outside of LINZs corporate network. I will however be opening a ticket with datacom this morning to resolve this so that QGIS plugins can be connect to the LINZ-Experimental server in the DaaS environment for testing (managed via linz/ds-dev-team/issues/74). 

In the mean time you can test the plugin against QGIS 2.18.16 in your desktop environment to sign off on the plugins new functionality and the proper functioning against the QGIS2.18 api. 

Using the MS Windows OS it is possible to install parallel instances of QGIS. This is handy for the below managing of different plugin versions as you can test against a QGIS instance (mostly) independent of you prod environment. 

THOUGH, do be careful as the QGIS plugins can be installed to several places including you common users directory where multiple versions of QGIS could access its plugins from. If this is the case when switch back from your testing work to BAU you will need to downgrade the plugin. Lets try avoid this... 


**Upgrade the AIMS QGIS Plugin version**
The plugin is available via the dev plugin server (I can email you the url). The plugins on this servers can be added by adding the plugin servers details to the QGIS plugin settings.  The AIMS plugin can then be installed/upgraded as usual in the QGIS plugin manager. (if you have not done this before I can support this). From memory by default this will see you writing to the common user directory. 

So, It is probably best you download the plugin from the github release (https://github.com/linz/QGIS-AIMS-Plugin/releases/tag/v1.1.0) and place this in the QGIS instance plugin dir (<QGIS install path>\apps\qgis\python\plugins). You will need to update the config file using this method via this [documentation](https://github.com/linz/AIMS-deployment#aims-qgis-plugin-development)

**Connecting the plugin to the test environment** 

(Apologies) You must use PUTTY to forward the LINZ-experimental servers ports, allowing you to interact with the test environment. the details how to do this can be found [here](https://github.com/linz/li-experimental-server/wiki/Connecting-to-LI--Experimental-Server-from-Windows). Again let me know if you want help configuring this. 

Before testing against this environment let me know and I can confirm it is up and suitable for conducting testing against.

**Comment by linzbjones** [comment]
Sent test plan to @SPlanzer and @dwsilk for review.

**Comment by SPlanzer** [comment]
@jason-bedford I have added test 18, 19 as related to the new release that enables parcel labeling  

for tests such as "During update, Level Value can not be deleted when Level Type exists " (~197) that have a result such as "Change appears on Review queue. Warning: Level Value has not been supplied" should these all have a corresponding test that upon fixing the issues the warning has been removed.  

"Group review queue" also needs more detail. This will need to involve getting the groups of different nature (parcel, meshblock, road and locality updates) on the queue. In the past [this test suite](https://github.com/linz/AIMS-SQL-interface-test) has been employed. We will need instructions for this. Alternatively the test data restored to the LI-Server could have these examples dumped for the restore and ready to go for the tester.

I also note "Group review queue" has no reference to editing individual items in the group. Its been a while, is this possible? 

Please also see the [release notes](https://github.com/linz/QGIS-AIMS-Plugin/releases/tag/v1.1.0). This outlines the changes to the plugin this have specific tests that will need to be performed in this round of testing to ensure the release meets expectations

**Comment by SPlanzer** [comment]
This is blocked by linz/ds-dev-team/issues/147 - Currently when installing a Plugin into QGIS (official version as part of DaaS), when logging out and starting the next DaaS session the plugin must be re-installed and configured. 

I have however started initial testing with the new AIMS release and QGIS 2.18 as installed in DaaS. Initial results are not good. When loading the plugin and panning / rendering AIMS features QGIS crashes. This is not however the case with QGIS installed (hacked - copying entire QGIS dir from another machine to personal DaaS dir) in DaaS outside an app silo (So far all initial test has in the same version of QGIS installed in this manner). - More details to come in issue
